packages/core/src/service/services/auth.service.ts

```diff
@@ -97,8 +97,10 @@ export class AuthService {
         if (ctx.session && ctx.session.activeOrderId) {
             await this.sessionService.deleteSessionsByActiveOrderId(ctx, ctx.session.activeOrderId);
         }
-        user.lastLogin = new Date();
-        await this.connection.getRepository(ctx, User).save(user, { reload: false });
+        if (user.roles || !this.configService.authOptions.disableLastLoginUpdate) {
+            user.lastLogin = new Date();
+            await this.connection.getRepository(ctx, User).save(user, { reload: false });
+        }
```

Comment:
```suggestion
        if (!this.configService.authOptions.disableLastLoginUpdate) {
            user.lastLogin = new Date();
            await this.connection.getRepository(ctx, User).save(user, { reload: false });
        }
```

I'm curious, why include `user.roles` in the condition? Especially if it specifically gets set above, doesnt that result in this always being cast to `true` since even empty arrays are `true` ?

Reply:
@alexisvigoureux could we please remove the `user.roles` condition? I think it is not necessary. If that's a way how do differentiate between a customer and an administrator, we should do it differently.